engine/engine/management/commands/continue_escalation.py

```diff
@@ -0,0 +1,133 @@
+from celery import uuid as celery_uuid
+from django.core.management import BaseCommand
+from django.db.models import Q
+from django.utils import timezone
+
+from apps.alerts.models import AlertGroup, AlertReceiveChannel
+from apps.alerts.tasks import escalate_alert_group, unsilence_task
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        group = parser.add_mutually_exclusive_group(required=True)
+
+        group.add_argument(
+            "-id", "--alert_group_ids", type=int, nargs="+", help="Alert group IDs to restart escalation for."
+        )
+        group.add_argument(
+            "-ppk", "--alert_group_ppk", type=str, nargs="+", help="Alert group public pks to restart escalation for."
+        )
+        group.add_argument(
+            "--all", action="store_true", help="Restart escalation for all alert groups with unfinished escalation."
+        )
+        parser.add_argument(
+            "-uns", "--unsilence_task", action="store_true", help="Restart unsilence task for selected alert groups."
+        )
+
+    def handle(self, *args, **options):
+        alert_group_ids = options["alert_group_ids"]
+        alert_group_ppk = options["alert_group_ppk"]
+        restart_all = options["all"]
+        restart_unsilence_task = options["unsilence_task"]
+
+        if restart_all:
+            self.stdout.write("Processing restart escalation for all active alert groups...")
+            alert_groups = AlertGroup.objects.filter(
+                ~Q(channel__integration=AlertReceiveChannel.INTEGRATION_MAINTENANCE),
+                ~Q(silenced=True, silenced_until__isnull=True),  # filter silenced forever alert_groups
+                Q(Q(is_escalation_finished=False) | Q(silenced_until__isnull=False)),
+                resolved=False,
+                acknowledged=False,
+                root_alert_group=None,
+            )
```

Comment:
do we have this same query elsewhere in the codebase? (it looks quite similar to the one used for `check_escalation_finished`).Wwe may want to consider deduplicating it/moving it to the [`AlertGroupQuerySet`](https://github.com/grafana/oncall/blob/dev/engine/apps/alerts/models/alert_group.py#L98) + adding some tests there.